router.go

```diff
@@ -51,8 +52,14 @@ type TreeMux struct {
 
 	// The default PanicHandler just returns a 500 code.
 	PanicHandler PanicHandler
+
 	// The default NotFoundHandler is http.NotFound.
 	NotFoundHandler func(w http.ResponseWriter, r *http.Request)
+
+	// The default OptionsHandler is nil. If set, the handler will run on any OPTIONS requests
+	// without an explicit match.
```

Comment:
Ok, I misunderstood the lifecycle and thought that it matched everything. I'll make the change.